src/main/kotlin/karazin/scala/users/group/week1/homework/program.kt

```diff
@@ -2,10 +2,34 @@ package karazin.scala.users.group.week1.homework
 
 import karazin.scala.users.group.week1.homework.adt.ErrorOr
 import karazin.scala.users.group.week1.homework.model.PostView
+import karazin.scala.users.group.week1.homework.services.getComments
+import karazin.scala.users.group.week1.homework.services.getPosts
+import karazin.scala.users.group.week1.homework.services.getUserProfile
+import karazin.scala.users.group.week1.homework.services.getLikes
+import karazin.scala.users.group.week1.homework.services.getShares
 
 object program {
 
-    fun getPostsViews(): ErrorOr<List<ErrorOr<PostView>>> = throw NotImplementedError()
-    fun getPostView(post: model.Post): ErrorOr<PostView> = throw NotImplementedError()
+    fun getPostsViews(): ErrorOr<List<PostView>> =
+             getUserProfile().flatMap { profile ->
+                getPosts(profile.userId).flatMap { posts ->
+                    ErrorOr.apply(posts.map { post -> getPostView(post) })
+                }
+            }
+
+
+    fun getPostView(post: model.Post): PostView =
```

Comment:
1. The changes should be done in the previous PR, not the new one.
2. I doubt that this code compiles (prove me wrong).
3. One more case with uncompilable code and you're failed.